### PR TITLE
Enhanced: Added State search filter

### PR DESCRIPTION
**Pull request by contemplate**
- ability to filter by State
- helpful for calculating state sales tax

### CODE REVIEW

**Comment by strangerstudios** [comment]
I like the idea, but I think it would work better as an addon. Or maybe a code gist that allowed you to do a search like bstate:PA. Are you a PMPro member? If you post to the forums, we can help you to work through that.

**Comment by contemplate** [comment]
Yep I'm a member!  Yeah I had tried to just make a copy of the Sales & Rev report for an extend plugin of sorts but then things got too complicated.  I'll post to the forum.

File: adminpages/reports/sales.php [hunk 1]
```diff
@@ -108,6 +108,11 @@ function pmpro_report_sales_page()
 		$l = intval($_REQUEST['level']);
 	else
 		$l = "";
+		
+	if(isset($_REQUEST['b_state']))
+		$s = strtoupper($_REQUEST['b_state']);
```
Comment (sc0ttkclark) [inline adminpages/reports/sales.php:113]:
```suggestion
		$s = sanitize_text_field( strtoupper( $_REQUEST['b_state'] ) );
```

**Comment by sc0ttkclark** [comment]
Closing based on previous comment from Jason.

**Pull request closed without merging**